subsys/tracing/ctf/ctf_top.c

```diff
@@ -274,24 +274,48 @@ void sys_trace_k_mutex_unlock_exit(struct k_mutex *mutex, int ret)
 }
 
 /* Timer */
-void sys_trace_k_timer_init(struct k_timer *timer, k_timer_expiry_t expiry_fn,
-			    k_timer_expiry_t stop_fn)
+void sys_trace_k_timer_init(struct k_timer *timer)
 {
+	ctf_top_timer_init(
+		(uint32_t)(uintptr_t)timer);
 }
 
 void sys_trace_k_timer_start(struct k_timer *timer, k_timeout_t duration,
 			     k_timeout_t period)
 {
+	ctf_top_timer_start(
+		(uint32_t)(uintptr_t)timer,
+		k_ticks_to_ms_floor32((uint32_t)duration.ticks),
```

Comment:
I don't think we should be rounding the ticks off to ms, us perhaps, but not ms.